"The truth value of an array with more than one element is ambiguous" loading np.array values

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
def test_construct_from_nparray_value() -> None:
    array_cell = np.array([2, 3])
    df = pl.DataFrame(np.array([[array_cell, 4]], dtype=object))
    assert df.dtypes == [pl.Object, pl.Object]
    to_numpy = df.to_numpy()
    assert to_numpy.shape == (1, 2)
    assert_array_equal(to_numpy[0][0], array_cell)
    assert to_numpy[0][1] == 4
```


### Log output

```shell
I added a temptative unit test to tests/unit/interop/test_interop.py

POLARS_VERBOSE=1 PYTHONPATH=. pytest tests/unit/interop/test_interop.py -k test_from_numpy_nparray_value
============================ test session starts ============================
platform linux -- Python 3.11.5, pytest-8.0.0, pluggy-1.4.0
rootdir: /home/dani/dev/python/polars/py-polars
configfile: pyproject.toml
plugins: hypothesis-6.97.4, xdist-3.5.0, cov-4.1.0
collected 151 items / 150 deselected / 1 selected                           

tests/unit/interop/test_interop.py F                                  [100%]

================================= FAILURES ==================================
_______________________ test_from_numpy_nparray_value _______________________
polars/datatypes/constructor.py:135: in numpy_type_to_constructor
    return _NUMPY_TYPE_TO_CONSTRUCTOR[dtype]  # type:ignore[index]
E   KeyError: <class 'numpy.object_'>

During handling of the above exception, another exception occurred:
tests/unit/interop/test_interop.py:497: in test_from_numpy_nparray_value
    df = pl.from_numpy(np.array([[array_cell, 4]], dtype=object))
polars/convert.py:505: in from_numpy
    return pl.DataFrame._from_numpy(
polars/dataframe/frame.py:551: in _from_numpy
    numpy_to_pydf(
polars/utils/_construction.py:1524: in numpy_to_pydf
    data_series = [
polars/utils/_construction.py:1525: in <listcomp>
    pl.Series(
polars/series/series.py:311: in __init__
    self._s = numpy_to_pyseries(
polars/utils/_construction.py:265: in numpy_to_pyseries
    constructor = numpy_type_to_constructor(values, dtype)
polars/datatypes/constructor.py:138: in numpy_type_to_constructor
    first_non_nan = next((v for v in values if v == v), None)
polars/datatypes/constructor.py:138: in <genexpr>
    first_non_nan = next((v for v in values if v == v), None)
E   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
========================== short test summary info ==========================
FAILED tests/unit/interop/test_interop.py::test_from_numpy_nparray_value - ValueError: The truth value of an array with more than one element is am...
===================== 1 failed, 150 deselected in 0.65s =====================
```


### Issue description

The error occurs when creating a DataFrame where a cell is a numpy array (eg pl.List dtype).
This is required for instance if you create a DataFrame were you split a column with `pl.col("name").str.split(", ")`, and you export the data with to_numpy, operate on it and create a new DatFrame with from_numpy`.
This worked on 0.20.5, but was broken by commit c732f26bbd0e60ac72f599a20b4d636913d0f00a where the `v == v ` fails for np.arrays
```
            first_non_nan = next((v for v in values if v == v), None)
```

### Expected behavior

A DataFrame with an object columns is created. See asserts in "reproducible example"

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.6
Index type:           UInt32
Platform:             Linux-6.5.0-9-generic-x86_64-with-glibc2.38
Python:               3.11.5 (main, Sep 20 2023, 13:23:03) [GCC 12.2.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          2.2.1
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.9.1
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.8.2
numpy:                1.26.3
openpyxl:             <not installed>
pandas:               2.2.0
pyarrow:              14.0.2
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>

```

</details>

--- a/py-polars/polars/datatypes/constructor.py
+++ b/py-polars/polars/datatypes/constructor.py
@@ -135,7 +135,9 @@ def numpy_type_to_constructor(
         return _NUMPY_TYPE_TO_CONSTRUCTOR[dtype]  # type:ignore[index]
     except KeyError:
         if len(values) > 0:
-            first_non_nan = next((v for v in values if v == v), None)
+            first_non_nan = next(
+                (v for v in values if isinstance(v, np.ndarray) or v == v), None
+            )
             if isinstance(first_non_nan, str):
                 return PySeries.new_str
             if isinstance(first_non_nan, bytes):

--- a/py-polars/tests/unit/interop/test_interop.py
+++ b/py-polars/tests/unit/interop/test_interop.py
@@ -476,6 +476,32 @@ def test_from_numpy() -> None:
         _ = pl.from_numpy(np.array(1))
 
 
+def test_from_numpy_array_value() -> None:
+    df = pl.DataFrame({"A": [[2, 3]]})
+    assert df.rows() == [([2, 3],)]
+    assert df.schema == {"A": pl.List(pl.Int64)}
+
+
+def test_construct_from_nparray_value() -> None:
+    array_cell = np.array([2, 3])
+    df = pl.DataFrame(np.array([[array_cell, 4]], dtype=object))
+    assert df.dtypes == [pl.Object, pl.Object]
+    to_numpy = df.to_numpy()
+    assert to_numpy.shape == (1, 2)
+    assert_array_equal(to_numpy[0][0], array_cell)
+    assert to_numpy[0][1] == 4
+
+
+def test_from_numpy_nparray_value() -> None:
+    array_cell = np.array([2, 3])
+    df = pl.from_numpy(np.array([[array_cell, 4]], dtype=object))
+    assert df.dtypes == [pl.Object, pl.Object]
+    to_numpy = df.to_numpy()
+    assert to_numpy.shape == (1, 2)
+    assert_array_equal(to_numpy[0][0], array_cell)
+    assert to_numpy[0][1] == 4
+
+
 def test_from_numpy_structured() -> None:
     test_data = [
         ("Google Pixel 7", 521.90, True),

